packages/components/src/dialog/button-bar.tsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import React, { isValidElement, cloneElement } from 'react';
+import type { ReactElement, ReactNode, FunctionComponent } from 'react';
+import classnames from 'classnames';
+
+export type BaseButton = {
+	action: string;
+	disabled?: boolean;
+	label: ReactNode;
+	className?: string;
+	additionalClassNames?: string;
+	isPrimary?: boolean;
+	onClick?: ( closeDialog: () => void ) => void;
+};
+
+export type Button = ReactElement | BaseButton;
+
+type Props = {
+	buttons: Button[];
+	baseClassName: string;
+	onButtonClick: ( button: BaseButton ) => void;
+};
+
+const ButtonBar: FunctionComponent< Props > = ( { buttons, baseClassName, onButtonClick } ) => {
+	if ( ! buttons ) {
+		return null;
+	}
+
+	return (
+		<div className={ baseClassName + '__action-buttons' }>
+			{ buttons.map( ( button, index ) => {
+				if ( isValidElement( button ) ) {
+					return cloneElement( button, { key: 'dialog-button-' + index } );
```

Comment:
Is using `index` for a key here [safe](https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318)? Can we use the `action` prop instead?

Reply:
The button doesn't have any `action` prop when it's a JSX element. The `buttons` array can be a mix of JSX elements and button configs:
```js
[
  <div>whatever</div>,
  { action: 'close', label: 'Close' },
  <button>click</button>
]
```
And only the configs have an `action` prop.

The `'dialog-button-'` prefix is redundant. A `key={ index }` prop is sufficient.

The `index` key should be perfectly safe in this case. In practice, the `buttons` array never changes.